core/src/main/scala/org/http4s/parser/Rfc3986Parser.scala

```diff
@@ -14,6 +14,8 @@ private[parser] trait Rfc3986Parser { this: Parser =>
 
   def charset: Charset
 
+  def UriTotal: Rule1[org.http4s.Uri] = rule { Uri ~ EOI }
```

Comment:
Sure. I can do that. Btw, the reason I created a separate rule is because adding `EOI` to the `Uri` rule makes unusable within another rule. I kind of think this is a design flaw of `parboiled2`, it seems like it should be possible to `run` a rule in either a mode where is parses some amount of input and another mode where it expects to consume all input. I don't know what the best "workaround" is, I guess for now it either way works as `Uri` is only ever used from the top level.